### DISCUSSION TITLE
Does aioble save service discovery for bonded reconnects

### DISCUSSION THREAD

**Discussion by brianreinhold** [micropython/micropython#13589] [Libraries & Drivers]
Does the aioble library save the results of service discovery for a bonded connection such that on a reconnect service discovery does not have to be redone? Many bonded devices on reconnect after encryption will send measurements directly and not wait for characteristics to be enabled.

**Comment by andrewleech** [comment]
No this isn't handled automatically. Support for "service changed" flag or "database hash" would need to be finished before this could be safely done I expect.

**Reply by brianreinhold** [reply]
As long as the application can save the necessary information to access the characteristics etc then it is okay. I have had to do that on some embedded platforms

I guess this is just a TODO for the library?
